src/components/routes/Booking/BookingPayment/BookingPayment.tsx

```diff
@@ -32,47 +39,66 @@ const BookingPayment = ({ history, match }: RouterProps) => (
         alert('Please select a payment option.');
         return <Redirect to={`/bookings/${match.params.id}/options`} />;
       }
+      const queryParams: QueryParams = parseQueryString(location.search);
+      const currency = Object.values(Currency).find(c => c === queryParams.currency);
+      const web3 = loadWeb3();
+      const pricePromise = !!currency && currency !== booking.currency ?
+        priceWithToken(web3.eth, currency, booking.guestTotalAmount) :
+        Promise.resolve(booking.guestTotalAmount);
       return (
-        <BookingPaymentContainer>
-          <BookingNavBar />
-          <AppConsumer>
-            {({ screenType }: AppConsumerProps) => {
-              if (screenType < ScreenType.TABLET) {
-                return (
-                  <div className="booking-payment-mobile-body">
-                    <TermsAndConditions houseRules={booking.listing.houseRules} />
-                    <div className="booking-payment-footer-container">
-                      <BookingPaymentBar booking={booking} />
-                    </div>
-                  </div>
-                );
-              }
-              return (
-                <div className="booking-payment-desktop-body">
-                  <div className="booking-payment-terms">
-                    <TermsAndConditions houseRules={booking.listing.houseRules} />
-                    <div className="booking-payment-button-container">
-                      <Button
-                        className="back-button"
-                        background="light"
-                        onClick={() => history.push(`/bookings/${booking.id}/options`)}
-                      >
-                        Go Back
-                      </Button>
-                      <BookingPaymentButton
-                        booking={booking}
-                        onSuccess={() => history.push(`/bookings/${booking.id}/receipt`)}
-                      />
+        <Async promise={pricePromise} then={price => {
+          const fromBee = (!!price && currency !== booking.currency) ?
+            ((value: number) => value * price * 1.01 / booking.guestTotalAmount) :
```

Comment:
1.01 ? is this error margin for approval?

Reply:
Yep, there are two cases where this matters:

1. When values in Uniswap exchanges change between price being queried and user submitting payment
2. When there is a loss of precision going back and forth to the JS `number` type

The second case could be addressed by keeping prices as strings in wei/dust, but that seemed like an uphill battle (much of the current code assumes numbers) and we'd still need to do something similar for the first case.